vendor/github.com/waku-org/go-waku/waku/v2/dnsdisc/enr.go

```diff
@@ -44,16 +44,19 @@ func init() {
 }
 
 // RetrieveNodes returns a list of multiaddress given a url to a DNS discoverable ENR tree
-func RetrieveNodes(ctx context.Context, url string, opts ...DNSDiscoveryOption) ([]DiscoveredNode, error) {
+func RetrieveNodes(ctx context.Context, url string, resolver dnsdisc.Resolver, opts ...DNSDiscoveryOption) ([]DiscoveredNode, error) {
```

Comment:
Instead of adding a new mandatory `resolver` parameter, this can be implemented by adding a new `func WithResolver(resolver *net.Resolver) DNSDiscoveryOption`  in this file. That way if someone does not need to use an specific resolver, they don't have to even worry about figuring out what value to pass in `resolver`, and you would not need to have a `dnsResolver` attribute in the `Waku` type